nano/node/network.cpp

```diff
@@ -891,32 +891,35 @@ void nano::tcp_message_manager::put_message (nano::tcp_message_item const & item
 {
 	{
 		nano::unique_lock<std::mutex> lock (mutex);
-		while (entries.size () > max_entries && !stopped)
+		while (entries.size () >= max_entries && !stopped)
 		{
 			condition.wait (lock);
 		}
 		entries.push_back (item_a);
 	}
-	condition.notify_all ();
+	condition.notify_one ();
 }
 
 nano::tcp_message_item nano::tcp_message_manager::get_message ()
 {
+	nano::tcp_message_item result;
 	nano::unique_lock<std::mutex> lock (mutex);
 	while (entries.empty () && !stopped)
 	{
 		condition.wait (lock);
 	}
 	if (!entries.empty ())
 	{
-		auto result (entries.front ());
+		result = std::move (entries.front ());
 		entries.pop_front ();
-		return result;
 	}
 	else
 	{
-		return nano::tcp_message_item{ std::make_shared<nano::keepalive> (), nano::tcp_endpoint (boost::asio::ip::address_v6::any (), 0), 0, nullptr, nano::bootstrap_server_type::undefined };
+		result = nano::tcp_message_item{ std::make_shared<nano::keepalive> (), nano::tcp_endpoint (boost::asio::ip::address_v6::any (), 0), 0, nullptr, nano::bootstrap_server_type::undefined };
 	}
+	lock.unlock ();
+	condition.notify_one ();
```

Comment:
The same issue as above, this could wake another thread waiting on get_message and not a producer.